sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -835,6 +835,15 @@ object SQLConf {
       .longConf
       .createWithDefault(250 * 1024 * 1024)
 
+  val HIVE_FILESOURCE_PARTITION_FILE_CACHE_TTL =
+    buildConf("spark.sql.hive.filesourcePartitionFileCacheTTL")
```

Comment:
Since the `FileStatusCache` implementation is independent of catalog impls, I personally think its better to avoid using `hive` in the config name.

Reply:
How about `spark.sql.catalog.fileStatusCacheTTL`? Then, describing more about when this config enabled?

Reply:
This is a nice-to-have feature, but we need to cover all the metadata cache, instead of adding TTL for each metadata cache.

Reply:
So, `spark.sql.hive.filesourcePartitionFileCacheTTL` simply mirrors the existing `spark.sql.hive.filesourcePartitionFileCacheSize` config. I think it makes sense for consistency, since they're applied to the same cache.

I don't mind dropping `hive`, but in this case it also makes sense to drop it from the `spark.sql.hive.filesourcePartitionFileCacheSize` and it's going to be a backwards-incompatible change.

What do you think?

Reply:
Ah, I see. But, I personally think its better to remove `hive` in that name, too, because of the same reason.

>  it's going to be a backwards-incompatible change.

Yea, since we must keep the compatibility, we need to move the current config  into `removedSQLConfigs` then add a new config name for that.

cc: @dongjoon-hyun @HyukjinKwon

Reply:
@maropu got it, will proceed with dropping `hive` from these two configs.

@gatorsmile could you elaborate a bit? Are you just referring to dropping the `hive` prefix or is there something else I'm missing? Thanks

Reply:
> @gatorsmile could you elaborate a bit? Are you just referring to dropping the hive prefix or is there something else I'm missing? Thanks

We have the cache mechanism for metadata-related items, e.g., 
 https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L138 IIUC he suggested its better for them to have the same TTL.

Reply:
@maropu @gatorsmile I've changed the config option to `spark.sql.metadataCacheTTL` and applied it to `FileStatusCache` as well as `SessionCatalog` in https://github.com/apache/spark/pull/28852/commits/28da5cfd39dba6a8319dd1cdfe39e51ed5cbdea5. What do you think? 

I'm happy to update `spark.sql.hive.filesourcePartitionFileCacheSize` (drop `hive` prefix), but it's clearly out of scope here, I could follow-up with another PR after this one is resolved.

Thanks again.

Reply:
Plz update the PR description, too?

> I'm happy to update spark.sql.hive.filesourcePartitionFileCacheSize (drop hive prefix), but it's clearly out of scope here, I could follow-up with another PR after this one is resolved.

Yea, the topic is out-of-scope in this PR. I'm not 100% sure that we need to rename it now (because the option has been used for a long time, since 2.1.1), but I think making another PR for that and discussing it sounds nice.

Reply:
@maropu right, updated